### PR TITLE
Flagged several string arrays; Added Copy method for Note

### PR DESCRIPTION
**Pull request by holmes301**

### CODE REVIEW

File: Note/Hold.cs [hunk 1]
```diff
@@ -83,24 +84,7 @@ public Hold(string noteType, int bar, int startTime, string key, int lastTime, i
...
-        LastTimeStamp = inTake.LastTimeStamp;
-        WaitLength = inTake.WaitLength;
-        WaitTickStamp = inTake.WaitTickStamp;
-        WaitTimeStamp = inTake.WaitTimeStamp;
-        CalculatedLastTime = inTake.CalculatedLastTime;
-        CalculatedLastTime = inTake.CalculatedLastTime;
-        TickBPMDisagree = inTake.TickBPMDisagree;
-        BPM = inTake.BPM;
-        BPMChangeNotes = inTake.BPMChangeNotes;
+        Note.Copy(this, inTake);
```
Comment (Neskol) [inline Note/Hold.cs:87]: Wouldn't it be better if use base constructor?

File: Note/Note.cs [hunk 2]
```diff
@@ -992,4 +992,30 @@ public override int GetHashCode()
...
+        copyTo.WaitLength = copyFrom.WaitLength;
+        copyTo.WaitTickStamp = copyFrom.WaitTickStamp;
+        copyTo.WaitTimeStamp = copyFrom.WaitTimeStamp;
+        copyTo.CalculatedLastTime = copyFrom.CalculatedLastTime;
+        copyTo.CalculatedLastTime = copyFrom.CalculatedLastTime;
+        copyTo.TickBPMDisagree = copyFrom.TickBPMDisagree;
+        copyTo.BPM = copyFrom.BPM;
+        copyTo.BPMChangeNotes = copyFrom.BPMChangeNotes;
+    }
 }
```
Comment (Neskol) [inline Note/Note.cs:1021]: Shouldn't be static: use this.Replace() instead

File: Note/Slide.cs [hunk 3]
```diff
@@ -141,24 +141,7 @@ public Slide(string noteType, int bar, int startTime, string key, int waitTime,
...
-        LastTimeStamp = inTake.LastTimeStamp;
-        WaitLength = inTake.WaitLength;
-        WaitTickStamp = inTake.WaitTickStamp;
-        WaitTimeStamp = inTake.WaitTimeStamp;
-        CalculatedLastTime = inTake.CalculatedLastTime;
-        CalculatedLastTime = inTake.CalculatedLastTime;
-        TickBPMDisagree = inTake.TickBPMDisagree;
-        BPM = inTake.BPM;
-        BPMChangeNotes = inTake.BPMChangeNotes;
+        Note.Copy(this, inTake);
```
Comment (Neskol) [inline Note/Slide.cs:144]: Consider base constructor

File: Note/SlideGroup.cs [hunk 4]
```diff
@@ -135,24 +135,7 @@ public override bool Update()
...
-            LastTimeStamp = inTake.LastTimeStamp;
-            WaitLength = inTake.WaitLength;
-            WaitTickStamp = inTake.WaitTickStamp;
-            WaitTimeStamp = inTake.WaitTimeStamp;
-            CalculatedLastTime = inTake.CalculatedLastTime;
-            CalculatedLastTime = inTake.CalculatedLastTime;
-            TickBPMDisagree = inTake.TickBPMDisagree;
-            BPM = inTake.BPM;
-            BPMChangeNotes = inTake.BPMChangeNotes;
+            Note.Copy(this, inTake);
```
Comment (Neskol) [inline Note/SlideGroup.cs:138]: Consider base constructor

File: Note/Tap.cs [hunk 5]
```diff
@@ -102,24 +102,8 @@ public Tap(string noteType, int bar, int startTime, string key, int specialEffec
...
-        LastTimeStamp = inTake.LastTimeStamp;
-        WaitLength = inTake.WaitLength;
-        WaitTickStamp = inTake.WaitTickStamp;
-        WaitTimeStamp = inTake.WaitTimeStamp;
-        CalculatedLastTime = inTake.CalculatedLastTime;
-        CalculatedLastTime = inTake.CalculatedLastTime;
-        TickBPMDisagree = inTake.TickBPMDisagree;
-        BPM = inTake.BPM;
-        BPMChangeNotes = inTake.BPMChangeNotes;
+        Note.Copy(this, inTake);
```
Comment (Neskol) [inline Note/Tap.cs:105]: Consider base constructor

Comment (holmes301) [reply Note/Tap.cs:105]: YOU consider base constructor

File: Note/Hold.cs (hunk 1 above)
Comment (holmes301) [reply Note/Hold.cs:87]: What base constructor?

**Review by Neskol — CHANGES_REQUESTED** [review]
Consider using non-static method for replacing information like`public void Replace(Note inTake)` and try to use base constructor of Notes

File: Note/Hold.cs (hunk 1 above)
Comment (Neskol) [reply Note/Hold.cs:87]: Constructor from Note()

**Pull request closed without merging**